src/Compiler/Microsoft.AspNetCore.Razor.Language/src/RazorCodeDocumentExtensions.cs

```diff
@@ -550,7 +561,16 @@ public override void VisitRazorDirective(RazorDirectiveSyntax node)
                 if (directiveContent != null && directiveContent.StartsWith(NamespaceDirective.Directive.Directive, StringComparison.Ordinal))
                 {
                     LastNamespaceContent = directiveContent.Substring(NamespaceDirective.Directive.Directive.Length).Trim();
-                    LastNamespaceLocation = node.GetSourceSpan(_source);
+                    var location = node.GetSourceSpan(_source);
+
+                    var offset = 1 + NamespaceDirective.Directive.Directive.Length + 1; // @ + namespace + " "
```

Comment:
What if there are multiple spaces (or other whitespae characters), like `@namespace     Test` - will that work?

Reply:
Yes, it will treat the 'user portion' as `    Test`. I'll update a test to include that.

Reply:
Added test, and we now generate even more correctly due to using the tokens.